sycl/include/CL/sycl/half_type.hpp

```diff
@@ -394,6 +394,81 @@ class half {
+  __SYCL_CONSTEXPR_HALF friend half operator op(const half &lhs,               \
+                                                const int &rhs) {              \
+    half rtn = lhs;                                                            \
+    rtn op_eq rhs;                                                             \
+    return rtn;                                                                \
+  }                                                                            \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const int &lhs,                \
+                                                const half &rhs) {             \
+    half rtn = rhs;                                                            \
+    rtn op_eq lhs;                                                             \
+    return rtn;                                                                \
+  }                                                                            \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const half &lhs,               \
+                                                const long &rhs) {             \
+    half rtn = lhs;                                                            \
+    rtn op_eq rhs;                                                             \
+    return rtn;                                                                \
+  }                                                                            \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const long &lhs,               \
+                                                const half &rhs) {             \
+    half rtn = rhs;                                                            \
+    rtn op_eq lhs;                                                             \
+    return rtn;                                                                \
+  }                                                                            \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const half &lhs,               \
+                                                const long long &rhs) {        \
+    half rtn = lhs;                                                            \
+    rtn op_eq rhs;                                                             \
+    return rtn;                                                                \
+  }                                                                            \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const long long &lhs,          \
+                                                const half &rhs) {             \
+    half rtn = rhs;                                                            \
+    rtn op_eq lhs;                                                             \
+    return rtn;                                                                \
+  }
+  OP(+, +=)
+  OP(-, -=)
+  OP(*, *=)
+  OP(/, /=)
```

Comment:
Can you also add OP(bool, ==)
  OP(bool, !=)
  OP(bool, <)
  OP(bool, >)
  OP(bool, <=)
  OP(bool, >=)?
Like this, we will have a more complete list similar to what we have for bfloat16 support currently.

Reply:
I have added the logical operators and expanded the test